src/mdesktopentry.cpp

```diff
@@ -243,32 +288,51 @@ QString MDesktopEntry::fileName() const
 
 bool MDesktopEntry::contains(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.contains(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.contains(key);
 }
 
 bool MDesktopEntry::contains(const QString &group, const QString &key) const
 {
-    return d_ptr->desktopEntriesMap.contains(group + '/' + key);
+    if (group != DesktopEntrySection) {
+        return false;
+    }
+    return d_ptr->desktopEntriesMap.contains(key);
 }
 
 QString MDesktopEntry::value(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.value(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.value(key);
 }
 
 QString MDesktopEntry::value(const QString &group, const QString &key) const
 {
-    return d_ptr->desktopEntriesMap.value(group + '/' + key);
+    if (group != DesktopEntrySection) {
+        return QString();
```

Comment:
I'm not aware of uses, but if aim is to enhance standard compliance on few details, would consider it good not to regress on others :) Also these group based query methods are nonsensical with support for only one group. 
Ditching desktopEntriesMap sounds good.